docs/topics/http/urls.txt

```diff
@@ -368,6 +368,41 @@ the following example is valid::
 In the above example, the captured ``"username"`` variable is passed to the
 included URLconf, as expected.
 
+Nested arguments
+================
+
+Regular expressions allow nested argument and Django will resolve them and
+pass them to the view. When reversing, Django will try to fill in all outer
+captured arguments, ignoring any nested captured arguments. Consider the
+following url patterns which optionally take a page argument::
+
+    from django.conf.urls import url
+
+    urlpatterns = [
+        url(r'blog/(page-(\d+)/)?$', blog_articles),
+        url(r'comments/(?:page-(?P<page_number>\d+)/)?$', comments),
+    ]
+
+Both patterns use nested arguments and will resolve. The ``blog_articles``
```

Comment:
In urlconf-lingo, resolving is "forwards" (URL to view+args), reversing is backwards (view+args to URL).